pkg/datapath/linux/probes/managed_neighbors.go

```diff
@@ -0,0 +1,84 @@
+package probes
+
+import (
+	"fmt"
+	"net"
+	"runtime"
+
+	"github.com/vishvananda/netlink"
+	"github.com/vishvananda/netns"
+)
+
+// HaveManagedNeighbors creates a netlink.Neigh entry with the flag NTF_EXT_MANAGED.
+func HaveManagedNeighbors() error {
+	ch := make(chan bool)
+	var err error
+
+	// In order to call HaveManagedNeighbors safely, it has to be started
+	// in a goroutine, so we can make sure the goroutine ends when the function exits.
+	// This gives us control over the lifecycle of the underlying thread.
+	go func() {
+		err = haveManagedNeighbors()
+		ch <- true
```

Comment:
IMO it's slightly more idiomatic to just close the channel:

```suggestion
		close(ch)
```